lib/ae_mdw_web/views/aex9_view.ex

```diff
@@ -61,17 +61,26 @@ defmodule AeMdwWeb.Views.Aex9ControllerView do
     do: transfer_to_map({sender_pk, recipient_pk, amount, call_txi, log_idx}, :aex9_transfer)
 
   def transfer_to_map({sender_pk, recipient_pk, amount, call_txi, log_idx}, :aex9_transfer) do
-    tx = call_txi |> Util.read_tx!() |> Format.to_map()
+    Model.tx(id: hash, block_index: {kbi, mbi}, time: micro_time) = Util.read_tx!(call_txi)
+    {_block_hash, _type, _signed_tx, tx_rec} = AeMdw.Node.Db.get_tx_data(hash)
+
+    contract_pk =
+      if elem(tx_rec, 0) == :contract_call_tx do
```

Comment:
We can use the `_type` from two lines above to avoid having to know the structure of `tx_rec` which should be an opaque term

Reply:
done